cypress-io/cypress
# breaking: add support for ressources with relative paths

- Closes [26725](https://github.com/cypress-io/cypress/issues/26725)

### Additional details


### Steps to test
create a cypress test that mounts a static ressource via a relative path, i.e. `<img src="./path/to/img.png" />` and check whether the request succeds

### How has the user experience changed?
It is now possible to distinguish the initial request for the index.html of the CT from relative asset requests (fonts/images/etc). This allows routing those requests to dev-server while retaining the relative paths.
Example: if a component test includes an image defined as follows: `<img src="./path/to/img.png" />` the load image request will be forwarded to the url `dev-server:port/__cypress/src/path/to/img.png` instead of `dev-server:port/__cypress/src/index.html`.

### PR Tasks

- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
